### PR TITLE
THEMES-704: Add Resizer V2

### PR DESCRIPTION
**Pull request by nschubach**
# TODO
- [ ] add appropriate dependencies to internal packages (`arc-themes-components`, `signing-service-content-source-block`)
- [x] `feeds-source-single-content-block` needs to be converted from resolve to fetch and add in the signing
- [x] `feeds-source-video-api-block` needs to be converted from resolve to fetch and add in the signing
- [ ] tests for `ans-feature-block` updated
- [ ] ~~tests for `feeds-source-collections-block` updated~~
- [ ] tests for `feeds-source-content-api-block` updated
- [ ] ~~tests for `feeds-source-content-api-by-day-block` updated~~
- [ ] ~~tests for `feeds-source-content-api-by-day2-block` updated~~
- [ ] ~~tests for `feeds-source-content-api-by-day3-block` updated~~
- [ ] tests for `rss-fbia-feature-block` updated
- [ ] tests for `rss-feature-block` updated
- [ ] tests for `rss-flipboard-feature-block` updated
- [ ] ~~tests for `rss-google-news-feature-block` updated~~
- [ ] tests for `rss-msn-feature-block` updated
- [ ] tests for `sitemap-video-feature-block` updated
- [ ] tests for `utils/content-elements` updated
- [ ] tests for `utils/promo-items` updated
- [ ] ~~tests for `utils/resizer` updated~~
- [x] fix `replaceAll()` CodeQL?

crossed out items have no tests...

### CODE REVIEW

File: blocks/feeds-source-content-api-block/sources/feeds-content-api.js [hunk 1]
```diff
@@ -1,84 +1,106 @@
...
+        if (ansFields.indexOf(i) !== -1) {
+          ansFields.splice(ansFields.indexOf(i), 1)
+        } else {
+          sourceExcludes.push(i)
+        }
       }
     })
-    if (sourceExclude.length) paramList._sourceExclude = sourceExclude.join(',')
+    if (sourceExcludes?.length)
+      paramList._sourceExclude = sourceExcludes.join(',')
```
Comment (rmbrntt) [inline blocks/feeds-source-content-api-block/sources/feeds-content-api.js]: paramList is undefined. In the diff on the left, 

```
  const paramList = {
    website: key['arc-site'],
    size: key['Feed-Size'] || '100',
    from: key['Feed-Offset'] || '0',
    sort: key.Sort || 'publish_date:desc',
  }
```
still called in yours though...

**Review by rmbrntt — COMMENTED** [review]
the amount of non-critical refactoring and shifting things around makes this challenging to track the critical logic, and with this repo's limited test coverage it's a bit 😬 

In retrospect it would've been ideal to have first have commits that did the resolve > fetch conversion and then separate commits or PRs for the minor refactors.

I noted at least one file where I believe there is actually missing code converted in the refactor (linting caught an unused param and a slice of code missing).

**Comment by rmbrntt** [comment]
I broke off a branch from this to do the following
1. Remove all arc-themes-blocks and arc-themes-components dependencies at the cost of duplication
2. Fix all tests (update snapshots, convert to axios). Need to change some of the test setup (seems the package bumps also introduced quirks there). 
3. I reverted the `feeds-content-api-source.js` back to the original develop version and re-converted it to v2 because there seemed to be bugs or unfinished work in `themes-704`, the diff was also challenging to trace through. I'm tempted to do the same for the collections source, I'd prefer to minimally touch these and only what is needed for a pure conversion. Later when test coverage is increased would we want to do this level of refactoring, but not coupled with this release.
4. Removed all those local package.json references. I'm not 100% why we need those again, would prefer we didn't as it is confusing and seemingly an uncommon way to work around something. I was able to use preconstruct locally and all refs worked fine between blocks and utils (after updating the workspace), unless you had those for linking for use in a feature pack?



https://github.com/WPMedia/feed-components/pull/666

It's base is set to this branch. 

If that gets approval, we'd still need to add any dependencies in the actual nested packages, like all the source blocks should define an axios in their package.json (should drop it from the top level, don't think it's needed - I did that when testing things and forgot til this moment).

Once that is done I believe we should release the utils. Then bump the version in blocks packages where they're using the utils. And then release a canary or beta so we can test it in a feature package with that tag.

File: utils/resizer/src/index.js [hunk 2]
```diff
@@ -1,19 +1,58 @@
...
+  ansImage,
 ) {
-  if (_url && typeof window === 'undefined') {
-    const Thumbor = require('thumbor-lite')
-    const thumbor = new Thumbor(resizerKey, resizerURL)
-    let imgSrc = _url.replace(/^http[s]?:\/\//, '').replace(' ', '%20')
-    if (imgSrc.includes('?')) {
-      imgSrc = imgSrc.replace('?', '%3F')
-    }
+  if (ansImage?.auth?.[RESIZER_TOKEN_VERSION]) {
```
Comment (rmbrntt) [inline utils/resizer/src/index.js:29]: Is this the right conditional?

Like we're checking if the auth token key exists on the ansImage under auth.RESIZER_TOKEN_VERSION, but if it is then below we end up adding it under auth, but directly as the auth value? 

Is this actually how we want to toggle to v2?

**Comment by rmbrntt** [comment]
add transforms back for 
feeds-source-content-api-by-day-block
feeds-source-content-api-by-day1-block
feeds-source-content-api-by-day2-block
feeds-source-content-api-by-day3-block

in #666 
feeds-content-api

**Review by rmbrntt — APPROVED** [review]

**Review by vgalatro — APPROVED** [review]

**Pull request merged**